cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceLifecycleContext.java

```diff
@@ -126,6 +127,16 @@ public String getNamespace() {
   }
 
   @Override
+  public void addLocalFile(URI uri) {
+    throw new UnsupportedOperationException("Local cache files are not supported in Mapper/Reducer tasks.");
```

Comment:
This is used in a Map/Reduce task, right? I thought we should only allow retrieving, and not adding in tasks. Adding should only be allowed in job setup/beforeSubmit?